feat: add animation to notification

aspnxdd opened the pull request:
Fixes #

## Changes

  - Adds an animation to the notification component.
  - Had to move onClose logic to the component, although it is extracted to a custom hook to handle the removal/destroyal of the component.
    
### Checks

* [x] Have you catchup your branch with the latest state in base?
* [x] Have you lint your code locally prior to submission?
* [x] Have you reviewed your proposed changes and removed debris?
* [x] Have you successfully ran due tests with your changes?

VmMad reviewed (changes requested):
I think there's some changes that could improve this visually. 
Reduce the timeout bar to half height, the opacity to anything between 0.175 and 0.2 would be better, also get the bar a little lower and apply a overflow hidden so it doesn't look like it's floating in the component.
1 thing more, if we close the notification there won't be any animation?

aspnxdd commented:
> I think there's some changes that could improve this visually. Reduce the timeout bar to half height, the opacity to anything between 0.175 and 0.2 would be better, also get the bar a little lower and apply a overflow hidden so it doesn't look like it's floating in the component. 1 thing more, if we close the notification there won't be any animation?

Please send me the values you think would improve this. I say this because if I reduce the height to 0.2rem (from 0.4rem), opacity to 0.2 and a margin-bottom of -0.05rem with overflow hidden it looks like this and is barely visible:
![image](https://user-images.githubusercontent.com/43625217/182653782-8fd81ada-3809-4b42-ad12-90137a5a6bfd.png)
And yes, I added the animation also when clicking the close button, although I had to move the onClose function inside the component.

VmMad commented on styles/globals.scss:
```diff
@@ -118,3 +118,43 @@ body {
+    margin-bottom:-0.05rem;
+    opacity: 0.2;
+    left: 0;
+    background-color: rgba(0, 0, 0, 0.253);
```
I think you carely can see it because you left `opacity:0.2;` and `background-color: rgba(0, 0, 0, 0.253);`  
```suggestion
    background-color: rgb(0 0 0);
```
with this or a lighter opacity it looks good!

aspnxdd replied:
You are right, I missed the rgba :joy_cat: thanks.

VmMad reviewed (approved):
Looks great!

cpl121 reviewed (approved):
Nice job, LGTM !

aspnxdd commented:
> I do not know if it is correct that when a cm is updated, the notification appears in the middle of the screen, instead of appearing at the bottom right of the screen @aspnxdd
> 
> ![image](https://user-images.githubusercontent.com/100352899/182902258-f0658ecf-444e-4437-a7db-73f649fcd2c3.png)

Fixed now. The position was absolute, so if we scrolled down, it seemed like it was in the middle of the screen. I changed it now to fixed so it scrolls down as well and it is kept at the bottom.

begonaalvarezd commented on styles/globals.scss:
```diff
@@ -118,3 +118,43 @@ body {
+    }
+}
+
+#notification::after {
+    content: '';
+    display: block;
+    width: 100%;
+    height: 0.2rem;
+    position: absolute;
+    bottom: 0;
+    margin-bottom:-0.05rem;
+    opacity: 0.2;
+    left: 0;
+    background-color: rgba(0, 0, 0);
+    z-index: 1;
+    transform-origin: left;
+    animation: shrink var(--data-timeout) ease-in-out forwards;
+}
```
This can be moved inside `#notification` 

```css
#notification {
    animation: slide-in 1s ease-in-out normal;
    overflow: hidden;
    
    &::after {
        content: '';
        display: block;
        width: 100%;
        height: 0.2rem;
        position: absolute;
        bottom: 0;
        margin-bottom:-0.05rem;
        opacity: 0.2;
        left: 0;
        background-color: rgba(0, 0, 0);
        z-index: 1;
        transform-origin: left;
        animation: shrink var(--data-timeout) ease-in-out forwards;
    }
}
```

begonaalvarezd commented on hooks/useNotification.tsx:
```diff
@@ -3,20 +3,41 @@ import { notificationState } from 'lib/recoil-store/atoms'
-const useNotification = () => {
+const styles = Object.freeze({
+    opacity: '0',
+    transform: 'translateY(-50%)',
```
I think without the vertical translate it would look a bit better, wdyt?

begonaalvarezd replied:
If you want a translate perhaps we could make it horizontal, from right as fadeIn and to right as fadeOut

begonaalvarezd replied:
I would also make the animation faster

begonaalvarezd commented:
🙏🏼
